src/main/java/org/myrobotlab/service/abstracts/AbstractSpeechRecognizer.java

```diff
@@ -570,18 +567,25 @@ public void unsetWakeWord() {
   }
 
   @Override
-  public void setLocale(String code) {
-    locale = new Locale(code);
+  public String getLanguage() {
+    return Runtime.getInstance().getLanguage();
   }
 
   @Override
-  public String getLanguage() {
-    return locale.getLanguage();
+  public Locale getLocale() {
+    return Runtime.getInstance().getLocale();
   }
 
   @Override
-  public Locale getLocale() {
-    return locale;
+  public Map<String, Locale> getLocales() {
+    return Locale.getAvailableLanguages();

```

Comment:
This is wrong, this should be the set of locales that the speech recognizer actually supports.  This should be abstract and force the subclasses to return the list of supported locales.

Reply:
agreed - fixed